### PR TITLE
update lead zero pattern for sub steps, re-work all CSS code

### PR DESCRIPTION
**Pull request by Christopher-O**

### CODE REVIEW

**Comment by duboisp** [comment]
Pre-approved upon review and quick local testing

File: méli-mélo/2021-05-steps/steps.css [hunk 1]
```diff
@@ -1,168 +1,120 @@
...
+    margin-left: -3em;
+    margin-top: -6px;
+}
+ol.lst-stps:not(.stps-strpd)>li, ol.lst-stps-sub:not(.stps-strpd)>li, .lst-stps:not(.stps-strpd) .lst-stps-itm {
+    margin-top: 20px;
+    min-height: 4em;
+    padding-left: 3.2em;
+    padding-right: 15px;
+}
+ol.lst-stps-sub:not(.stps-strpd)>li, .lst-stps:not(.stps-strpd) .lst-stps-sub .lst-stps-itm {
```
Comment (duboisp) [inline méli-mélo/2021-05-steps/steps.css:37]: This CSS selector are not matching any element in the demo page:

```css
.lst-stps:not(.stps-strpd) .lst-stps-sub .lst-stps-itm
```
I think that demo was removed. I can see it was used before

File: méli-mélo/2021-05-steps/steps.css [hunk 2]
```diff
@@ -1,168 +1,120 @@
...
+    min-height: 3em;
+    padding-left: 2.6em;
+}
+ol.lst-stps>li:before, ol.lst-stps>li ol.lst-stps-sub>li:before, .lst-stps .lst-stps-itm:before {
```
Comment (duboisp) [inline méli-mélo/2021-05-steps/steps.css:41]: This CSS selector are not matching any element in the demo page:

```css
.lst-stps .lst-stps-itm:before
```
I think that demo was removed. I can see it was used before

File: méli-mélo/2021-05-steps/steps.css [hunk 3]
```diff
@@ -1,168 +1,120 @@
...
+    font-weight: 600;
+    line-height: 2;
+    margin-left: -3.2em;
+    margin-right: 10px;
+    margin-top: -8px;
+    position: relative;
+    text-align: center;
+    width: 2em;
+}
+ol.lst-stps:not(.ld-zr)>li:before, ol.lst-stps:not(.ld-zr)>li ol.lst-stps-sub>li:before, .lst-stps:not(.ld-zr) .lst-stps-itm:before {
```
Comment (duboisp) [inline méli-mélo/2021-05-steps/steps.css:56]: This CSS selector are not matching any element in the demo page:

```css
.lst-stps:not(.ld-zr) .lst-stps-itm:before
```

I think that demo was removed. I can see it was used before

File: méli-mélo/2021-05-steps/steps.css [hunk 4]
```diff
@@ -1,168 +1,120 @@
...
+    background-color: #fff;
+    -webkit-border-radius: 50%;
+    -moz-border-radius: 50%;
+    border-radius: 50%;
+}
+.lst-stps.ld-zr .lst-stps-sub .lst-stps-itm:before {
```
Comment (duboisp) [inline méli-mélo/2021-05-steps/steps.css:62]: This CSS selector are not matching any element in the demo page:

```css
.lst-stps.ld-zr .lst-stps-sub .lst-stps-itm:before
```

I think that demo was removed. I can see it was used before

**Review by duboisp — CHANGES_REQUESTED** [review]
I completed a partial review

Can you remove the CSS selector that are not useful anymore?

If you need it for offering a transition for your experimental page, then create an example in another page that show the use of those deprecated CSS selector.

**Comment by duboisp** [comment]
> @duboisp just to clarify, you would like me to create a deprecated design page that showcases the old block element code that we are recommending not to be used anymore?

Is there any use, in the scope of your experimentation, to keep the deprecated CSS code? If not, remove it. If so then yes you can create a deprecated design page. 

Assume the scope of your experimentation is strictly limited to the content that you are already aware of it.

File: méli-mélo/2021-05-steps/steps.css [hunk 5]
```diff
@@ -1,168 +1,120 @@
...
+    counter-reset: subitem;
+    padding-left: 0px;
+}
+ol.lst-stps>li ol.lst-stps-sub>li:before, .lst-stps .lst-stps-sub .lst-stps-itm:before {
+    counter-increment: subitem;
+    content: counter(item) "" counter(subitem, lower-alpha) "";
+    margin-left: -3em;
+    margin-top: -6px;
+}
+ol.lst-stps:not(.stps-strpd)>li, ol.lst-stps-sub:not(.stps-strpd)>li, .lst-stps:not(.stps-strpd) .lst-stps-itm {
```
[the remark this answers is not in the document]
Comment (Christopher-O) [reply méli-mélo/2021-05-steps/steps.css]: outdated CSS removed

File: méli-mélo/2021-05-steps/steps.css (hunk 2 above)
Comment (Christopher-O) [reply méli-mélo/2021-05-steps/steps.css]: outdated CSS removed.

File: méli-mélo/2021-05-steps/steps.css (hunk 4 above)
Comment (Christopher-O) [reply méli-mélo/2021-05-steps/steps.css]: outdated CSS removed.

File: méli-mélo/2021-05-steps/steps.css (hunk 1 above)
Comment (Christopher-O) [reply méli-mélo/2021-05-steps/steps.css]: removed outdated CSS.

**Comment by GormFrank** [comment]
Pre-approved upon successful testing/review from @duboisp

**Review by duboisp — APPROVED** [review]
Tested locally, work as expected,

There is one little comment about an edge case with the selector "is" but that could be fixed later.

**Pull request merged**